### ISSUE TITLE
Provide a prop on `calcite-input-time-picker` for accessible lable (like `label`)

### ISSUE THREAD

**Issue by abp6318** [Esri/calcite-design-system#12153]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

There is no `label` prop provided, and attaching an `aria-label` or referencing another element with `aria-labelledby` results in aXe dev tool violations.

### Expected Behavior

Attaching an `aria-label` to the component should make it accessible and reduce aXe violations

### Reproduction Sample

https://codepen.io/AaronPutterman/pen/EaaMjbX?editors=1010

### Reproduction Steps

1. See repro sample
2. Open aXe dev tools in your browser
3. Run a page scan
4. See the two (possibly more) violations that appear -- one for "accessible name" and one for "permitted aria attributes"

### Reproduction Version

v3.1 @esri/calcite-components

### Priority impact

impact - p2 - want for an upcoming milestone

### Impact

Inputs not having accessible names impacts users using screen readers

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Hub

**Comment by jcfranco** [comment]
Related https://github.com/Esri/calcite-design-system/issues/7993.

**Comment by geospatialem** [comment]
@eriklharper It looks like the placement of the `label` needs to fit the `input`, rather than the `combobox` role per [the MDN recommendation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/combobox_role#:~:text=Every%20combobox%20must%20have%20an%20accessible%20name.%20If%20using%20an%20%3Cinput%3E%20element%2C%20the%20accessible%20name%20should%20come%20from%20the%20associated%20%3Clabel%3E.%20If%20not%2C%20if%20an%20appropriate%20label%20is%20visible%20in%20the%20content%2C%20provide%20the%20name%20via%20aria%2Dlabelledby.). Neither JAWS or NVDA are receiving context, so we may need to adjust the placement to [line 576](https://github.com/Esri/calcite-design-system/blob/24a0dfe10fd870b1dcfe07924f6a90a09cda404d/packages/calcite-components/src/components/input-time-picker/input-time-picker.tsx#L576) from [line 567](https://github.com/Esri/calcite-design-system/blob/24a0dfe10fd870b1dcfe07924f6a90a09cda404d/packages/calcite-components/src/components/input-time-picker/input-time-picker.tsx#L567).

Updating the status to `2 - in development`.
